re2.gemspec

```diff
@@ -1,8 +1,14 @@
+begin
+  require File.expand_path(File.join(File.dirname(__FILE__), "lib/re2/version"))
+rescue LoadError
+  puts "WARNING: Could not load RE2::VERSION"
+end
```

Comment:
As I recall, the `test-gem-install` line in https://github.com/mudge/re2/pull/76/commits/bfc359abb21c6cabf1482d3bf236192eac1202c0#diff-a229f587b00b7b78458a1dfaeabed7195dff0f58cbca9bba0e2fd4766017fb1aR32 removes the `lib` directory, so the `rake spec` line fails to load the version. I suppose this could be fixed by running the specs inside the installed gem directory instead of the current source dir? This was taken from Nokogiri.

Reply:
I'm trying to get this to work:

```diff
diff --git a/scripts/test-gem-install b/scripts/test-gem-install
index ef0317e..3783967 100755
--- a/scripts/test-gem-install
+++ b/scripts/test-gem-install
@@ -28,5 +28,6 @@ fi

 bundle install --local || bundle install

-rm -rf lib ext # ensure we don't use the local files
+cd $(dirname `gem which re2`)
+cd ..
 rake spec
stanhu@stanhu-gce:~/github/clean/re2$ git diff
diff --git a/re2.gemspec b/re2.gemspec
index 71c91de..cba9537 100644
--- a/re2.gemspec
+++ b/re2.gemspec
@@ -1,14 +1,10 @@
-begin
-  require File.expand_path(File.join(File.dirname(__FILE__), "lib/re2/version"))
-rescue LoadError
-  puts "WARNING: Could not load RE2::VERSION"
-end
+require_relative 'lib/re2/version'

 Gem::Specification.new do |s|
   s.name = "re2"
   s.summary = "Ruby bindings to re2."
   s.description = 'Ruby bindings to re2, "an efficient, principled regular expression library".'
-  s.version = defined?(RE2::VERSION) ? RE2::VERSION : "0.0.0"
+  s.version = RE2::VERSION
   s.authors = ["Paul Mucur"]
   s.homepage = "https://github.com/mudge/re2"
   s.extensions = ["ext/re2/extconf.rb"]
@@ -18,13 +14,15 @@ Gem::Specification.new do |s|
     "dependencies.yml",
     "ext/re2/extconf.rb",
     "ext/re2/re2.cc",
+    "Gemfile",
     "lib/re2.rb",
     "lib/re2/scanner.rb",
     "lib/re2/string.rb",
     "lib/re2/version.rb",
     "LICENSE.txt",
     "README.md",
-    "Rakefile"
+    "Rakefile",
+    "re2.gemspec"
   ]
   s.test_files = [
     "spec/spec_helper.rb",
diff --git a/scripts/test-gem-install b/scripts/test-gem-install
index ef0317e..3783967 100755
--- a/scripts/test-gem-install
+++ b/scripts/test-gem-install
@@ -28,5 +28,6 @@ fi

 bundle install --local || bundle install

-rm -rf lib ext # ensure we don't use the local files
+cd $(dirname `gem which re2`)
+cd ..
 rake spec
```

For some reason, it's not working yet. Semes like a `require 're2'` might be missing somewhere:

```
root@9abc1db3e04e:/usr/local/bundle/gems/re2-1.7.0# bundle exec rake spec
WARNING: rake-compiler found compiled files in 'ext/re2' directory. Please remove them.
/usr/local/bin/ruby -I/usr/local/bundle/gems/rspec-core-3.12.2/lib:/usr/local/bundle/gems/rspec-support-3.12.1/lib /usr/local/bundle/gems/rspec-core-3.12.2/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb

An error occurred while loading ./spec/re2/match_data_spec.rb.
Failure/Error:
  RSpec.describe RE2::MatchData do
    describe "#to_a" do
      it "is populated with the match and capturing groups" do
        a = RE2::Regexp.new('w(o)(o)').match('woo').to_a
        expect(a).to eq(["woo", "o", "o"])
      end

      it "populates optional capturing groups with nil if they are missing" do
        a = RE2::Regexp.new('(\d?)(a)(b)').match('ab').to_a
        expect(a).to eq(["ab", nil, "a", "b"])

NameError:
  uninitialized constant RE2::MatchData

  RSpec.describe RE2::MatchData do
                    ^^^^^^^^^^^
# ./spec/re2/match_data_spec.rb:2:in `<top (required)>'
```

Reply:
Are you including the [`.rspec`](https://github.com/mudge/re2/blob/main/.rspec) file which requires `spec_helper` and, in turn, `re2`?

Reply:
Aha, that was it. Thanks!